Makefile

```
@@ -23,6 +23,12 @@ LDFLAGS_COMMON := -X main.gitCommit=$(COMMIT) -X main.version=$(VERSION)
 
 GOARCH := $(shell $(GO) env GOARCH)
 
+# Allow Go 1.22.4+ to compile nsenter. This is ugly.
+GO122PATCH := $(shell $(GO) version | sed -n 's/.* go1\.22.\([0-9]*\) .*/\1/p')
+ifeq ($(shell test $(GO122PATCH) -ge 4 && echo yes), yes)
+	export CGO_CPPFLAGS+="-D GO1_22_WORKS=1"
```

Super nit, we are not using c++, so CFLAGS seem more appropriate (go passes both to C programs, though)

```suggestion
	export CGO_CFLAGS+="-D GO1_22_WORKS=1"
```

---

I guess you're mixing up CPPFLAGS (for C PreProcessor) and CXXFLAGS (C++).

---

Oh, yes, sorry. `.cpp` being used as file extension for c++ projects really confuses me with the CPPFLAGS :-D